include/fastdds/rtps/history/ReaderHistory.h

```diff
@@ -132,6 +138,38 @@ class ReaderHistory : public History
     RTPS_DllAPI void do_release_cache(
             CacheChange_t* ch) override;
 
+    template<typename Pred>
+    inline bool remove_changes_with_pred(
+            Pred pred)
+    {
+        assert(nullptr != mp_reader);
+        assert(nullptr != mp_mutex);
+
+        // Lock scope
+        std::vector<CacheChange_t*> changes_to_remove;
```

Comment:
The use of a std::vector here breaks zero allocation policy.

Reply:
Refactored into using std::remove_if